src/api/c/shift.cpp

```diff
@@ -18,12 +18,12 @@ using af::dim4;
 using namespace detail;
 
 template<typename T>
-static inline af_array shift(const af_array in, const af::dim4 &sdims)
+static inline af_array shift(const af_array in, const int sdims[4])
 {
     return getHandle(shift<T>(getArray<T>(in), sdims));
 }
 
-af_err af_shift(af_array *out, const af_array in, const af::dim4 &sdims)
+af_err af_shift(af_array *out, const af_array in, const int sdims[4])
```

Comment:
It is passing all four values.
On May 18, 2015 8:51 PM, "Umar Arshad" <notifications@github.com> wrote:

> In src/api/c/shift.cpp
> <https://github.com/arrayfire/arrayfire/pull/684#discussion_r30561064>:
>
> >  {
> >      return getHandle(shift<T>(getArray<T>(in), sdims));
> >  }
> >
> > -af_err af_shift(af_array *out, const af_array in, const af::dim4 &sdims)
> > +af_err af_shift(af_array *out, const af_array in, const int sdims[4])
>
> sdims should be a pointer. This is equivalent but it implies that it is
> passing all of the values.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/arrayfire/arrayfire/pull/684/files#r30561064>.
>